File: lib/java/src/main/java/io/github/brenbar/uapi/internal/validation/ValidateResult.java
package io.github.brenbar.uapi.internal.validation;

import static io.github.brenbar.uapi.internal.validation.MapValidationFailuresToInvalidFieldCases.mapValidationFailuresToInvalidFieldCases;

import java.util.List;

import io.github.brenbar.uapi.UApiError;
import io.github.brenbar.uapi.internal.types.UUnion;

public class ValidateResult {
    public static void validateResult(UUnion resultUnionType, Object errorResult) {
        final var newErrorResultValidationFailures = resultUnionType.validate(
                errorResult, null, null, List.of(), List.of());
        if (!newErrorResultValidationFailures.isEmpty()) {
            throw new UApiError(
                    "Failed internal uAPI validation: "
                            + mapValidationFailuresToInvalidFieldCases(newErrorResultValidationFailures));
        }
    }
}

File: tool/uapicodegen/uapicodegen/uapi/internal/validation/ValidateResult.py
from typing import TYPE_CHECKING
from ...UApiError import UApiError

if TYPE_CHECKING:
    from ...internal.types.UUnion import UUnion


def validate_result(result_union_type: 'UUnion', error_result: object) -> None:
    from ...internal.validation.MapValidationFailuresToInvalidFieldCases import map_validation_failures_to_invalid_field_cases
    from ...internal.validation.ValidateContext import ValidateContext

    new_error_result_validation_failures = result_union_type.validate(
        error_result, [], ValidateContext(None, None)
    )
    if new_error_result_validation_failures:
        raise UApiError(
            f"Failed internal uAPI validation: {map_validation_failures_to_invalid_field_cases(new_error_result_validation_failures)}"
        )
